### PR TITLE
Stable ordering when pretty printing, deserializing aliases.

### PR DESCRIPTION
**Pull request by nishantjr**
This fixes non-deterministic tests.

### CODE REVIEW

File: pyk/src/pyk/kcfg.py [hunk 1]
```diff
@@ -50,6 +51,11 @@ def id(self) -> str:
         def to_dict(self) -> Dict[str, Any]:
             return {'id': self.id, 'term': self.cterm.term.to_dict()}
 
+        def __lt__(self, other):
+            if not isinstance(other, KCFG.Node):
+                return NotImplemented
+            return self.id < other.id
+
```
Comment (ehildenb) [inline pyk/src/pyk/kcfg.py:57]: Can't this be enforced with our type-checker?

```suggestion
        def __lt__(self, other: KCFG.Node) -> bool:
            return self.id < other.id

```

File: pyk/src/pyk/kcfg.py [hunk 2]
```diff
@@ -58,6 +64,11 @@ class EdgeLike(ABC):
         def pretty_print(self, kprint: KPrint) -> List[str]:
             assert False, 'Must be overridden'
 
+        def __lt__(self, other):
+            if not isinstance(other, KCFG.EdgeLike):
+                return NotImplemented
+            return (self.source, self.target) < (other.source, other.target)
+
```
Comment (ehildenb) [inline pyk/src/pyk/kcfg.py:70]: Same here.

File: pyk/src/pyk/kcfg.py (hunk 1 above)
Comment (nishantjr) [reply pyk/src/pyk/kcfg.py:57]: That's a `return`, not `raises`. I think python will call the other classes implementation to see if that works.

File: pyk/src/pyk/kcfg.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import json
 from abc import ABC, abstractmethod
+from collections import OrderedDict
```
Comment (tothtamas28) [inline pyk/src/pyk/kcfg.py]: `dict` is insertion ordered as of Python 3.7, so this is probably not necessary.

File: pyk/src/pyk/kcfg.py [hunk 4]
```diff
@@ -285,8 +296,9 @@ def pretty_print(self, kprint: KPrint) -> List[str]:
         def _print_subgraph(indent: str, curr_node: KCFG.Node, prior_on_trace: List[KCFG.Node]) -> List[str]:
             ret: List[str] = []
 
+            edges_from = sorted(self.edge_likes(source_id=curr_node.id))
             if curr_node in processed_nodes:
-                if len(self.edge_likes(source_id=curr_node.id)) == 0:
+                if len(edges_from) == 0:
```
Comment (tothtamas28) [inline pyk/src/pyk/kcfg.py]:
```suggestion
                if not edges_from:
```

File: pyk/src/pyk/kcfg.py [hunk 5]
```diff
@@ -323,11 +335,11 @@ def _print_subgraph(indent: str, curr_node: KCFG.Node, prior_on_trace: List[KCFG
             return ret
 
         ret = []
-        init = self.init
+        init = sorted(self.init)
         while init:
             ret.append(self.node_short_info(init[0]))
             ret.extend(_print_subgraph('', init[0], [init[0]]))
-            init = [node for node in self.nodes if node not in processed_nodes]
+            init = [node for node in sorted(self.nodes) if node not in processed_nodes]
```
Comment (tothtamas28) [inline pyk/src/pyk/kcfg.py]:
```suggestion
            init = sorted(node for node in self.nodes if node not in processed_nodes)
```

**Review by tothtamas28 — APPROVED** [review]

**Pull request merged**